File: examples/perl/scripts/import.pl
#!/usr/bin/perl

use strict;
use warnings;

use File::Copy;
use File::Basename;
use VDI;

#
# Utilities
#

sub printWarning {
  my $line = $_[0];

  print STDOUT "$line\n";
}

#
# Parameter Handling
#

my ($inputDir, $outputDir) = @ARGV;

if (not defined $inputDir) {
  VDI::logMessage("Input directory was not specified");
  exit VDI::EXIT_CODE_UNEXPECTED_ERROR;
}
if (not defined $outputDir) {
  VDI::logMessage("Output directory was not specified");
  exit VDI::EXIT_CODE_UNEXPECTED_ERROR;
}

VDI::verifyDir($inputDir);
VDI::verifyDir($outputDir);

#
# Script body
#

VDI::logMessage("Here's a log message that will be printed to STDERR");

my @inputFiles = <$inputDir/*>;
foreach my $file (@inputFiles) {
  VDI::logMessage("Copying $file to $outputDir");
  printWarning("Here's a validation warning about $file that will be printed to STDOUT");
  copy($file, $outputDir . "/" . basename($file));
}

File: examples/python/scripts/import.py
#!/usr/bin/env python3

import os
import shutil
import sys
import vdi

#
# Utilities
#


def print_warning(msg):
    print(msg)


#
# Parameter Handling
#

if len(sys.argv) < 3:
    vdi.log_message("called with too few arguments, callers must provide an input and output directory")
    sys.exit(vdi.EXIT_CODE_UNEXPECTED_ERROR)

vdi.verify_dir(sys.argv[1])
vdi.verify_dir(sys.argv[2])

INPUT_DIR = sys.argv[1]
OUTPUT_DIR = sys.argv[2]

#
# Script Body
#

vdi.log_message("Here's a log message that will be printed to STDERR")

for file in os.listdir(INPUT_DIR):
    vdi.log_message(f"copying {file} to {OUTPUT_DIR}")
    print_warning(f"Here's a validation warning about {file} that will be printed to STDOUT")
    shutil.copy2(os.path.join(INPUT_DIR, file), os.path.join(OUTPUT_DIR, file))
